examples/embeddable_examples/public/react_embeddables/search/search_react_embeddable.tsx

```diff
@@ -24,54 +26,39 @@ export const getSearchEmbeddableFactory = (services: Services) => {
     deserializeState: (state) => {
       return state.rawState as State;
     },
-    buildEmbeddable: async (state, buildApi) => {
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const {
+        appliedTimeRange$,
+        cleanupTimeRange,
+        serializeTimeRange,
+        timeRangeApi,
+        timeRangeComparators,
+      } = initializeTimeRange(state, parentApi as Partial<PublishesTimeRange>);
       const defaultDataView = await services.dataViews.getDefaultDataView();
-      const timeRange$ = new BehaviorSubject<TimeRange | undefined>(state.timeRange);
       const dataViews$ = new BehaviorSubject<DataView[] | undefined>(
         defaultDataView ? [defaultDataView] : undefined
       );
       const dataLoading$ = new BehaviorSubject<boolean | undefined>(false);
-      function setTimeRange(nextTimeRange: TimeRange | undefined) {
-        timeRange$.next(nextTimeRange);
-      }
 
       const api = buildApi(
         {
+          ...timeRangeApi,
           dataViews: dataViews$,
-          timeRange$,
-          setTimeRange,
           dataLoading: dataLoading$,
           serializeState: () => {
             return {
               rawState: {
-                timeRange: timeRange$.value,
+                ...serializeTimeRange(),
```

Comment:
nit: If it's the only thing that's being serialized, we don't need the spread here.

Reply:
I think we should be consistent between initializers. The consumer does not know that there is only one item serialized. If some initializers require spread and others don't then how will consumers know how to use them. I will leave for consistency with titles initializers unless you feel strongly.

Reply:
I was actually just recommending a very small change on the consumer side here. Probably should've included an example, but I meant that it would be simpler to do:

``` typescript
rawState: serializeTimeRange(),
```

rather than

``` typescript
rawState: { ...serializeTimeRange() },
```

Reply:
thanks for the clarification. That makes sense. I can open a PR to make this change